akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala

```diff
@@ -941,35 +940,69 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef) extends Actor with
   /**
    * Initiates a new round of gossip.
    */
-  def gossip(): Unit = {
-
+  def gossip(): Unit =
     if (!isSingletonCluster) {
-      val localGossip = latestGossip
+      val gossipTargets =
+        // TODO break this two methods out and unit test?
+        if (latestGossip.isMultiDc) multiDcGossipTargets()
+        else localDcGossipTargets()
+
+      selectRandomNode(gossipTargets) match {
```

Comment:
a slightly other way would be, for each gossip round:

* decide own DC or other DC
  * if own => as before
  * if other => select one of the other DCs with equal probably
    * find the CrossDcConnections oldest for that DC, and then pick one random of them

I'm not sure what is best. If some DC contains many more nodes than another this approach might be better for spreading updates to all DCs quickly?